python/ray/data/tests/test_dataset_formats.py

```diff
@@ -409,6 +409,117 @@ def test_read_tfrecords(ray_start_regular_shared, tmp_path):
     assert list(df["bytes_list"]) == [[b"abc", b"1234"]]
 
 
+def test_write_tfrecords(ray_start_regular_shared, tmp_path):
```

Comment:
Nice explicit test! Using `tf.data.TFRecordDataset` as the ground-truth sounds good to me.

We should also add a "read our own writes" round-trip test, as we do for other I/O layers (e.g. [Parquet](https://github.com/ray-project/ray/blob/af4cbd40f4010f8151a8afa210ee2a3815fb96f0/python/ray/data/tests/test_dataset_parquet.py#L866-L898)), where we create a Dataset from in-memory data, write it to storage in the file format, read it back, and compare with the original in-memory data.

Reply:
+1 to add a "read our own writes" test.